api/opds.py

```diff
@@ -938,8 +921,163 @@ def add_authentication_document_link(self, feed_obj):
             )
         )
 
+class SharedCollectionAnnotator(CirculationManagerAnnotator):
+
+    def __init__(self, collection, lane,
+                 active_loans_by_work={}, active_holds_by_work={},
+                 active_fulfillments_by_work={},
+                 test_mode=False,
+    ):
+        super(SharedCollectionAnnotator, self).__init__(lane, active_loans_by_work=active_loans_by_work,
+                                                        active_holds_by_work=active_holds_by_work,
+                                                        active_fulfillments_by_work=active_fulfillments_by_work,
+                                                        test_mode=test_mode)
+        self.collection = collection
+
+    def top_level_title(self):
+        return self.collection.name
+
+    def default_lane_url(self):
+        return self.feed_url(None, default_route='crawlable_collection_feed')
+
+    def feed_url(self, lane, facets=None, pagination=None, default_route='feed'):
+        extra_kwargs = dict(collection_name=self.collection.name)
+        return super(SharedCollectionAnnotator, self).feed_url(lane, facets, pagination, default_route, extra_kwargs)
+
+    def annotate_work_entry(self, work, active_license_pool, edition, identifier, feed, entry):
+        updated = None
+        if isinstance(self.lane, CrawlableCollectionBasedLane) and isinstance(work, BaseMaterializedWork):
+            date_fields = [work.last_update_time, work.first_appearance, work.availability_time]
+            updated = max([date for date in date_fields if date is not None])
+
+        super(SharedCollectionAnnotator, self).annotate_work_entry(
+            work, active_license_pool, edition, identifier, feed, entry, updated
+        )
+                    
+    def acquisition_links(self, active_license_pool, active_loan, active_hold, active_fulfillment,
+                          feed, identifier):
+        """Generate a number of <link> tags that enumerate all acquisition methods."""
+        links = super(SharedCollectionAnnotator, self).acquisition_links(
+            active_license_pool, active_loan, active_hold, active_fulfillment, feed, identifier)
+
+        info_links = []
+        if active_loan:
+            url = self.url_for(
+                'shared_collection_loan_info',
+                collection_name=self.collection.name,
+                loan_id=active_loan.id,
+                _external=True)
+            kw = dict(href=url, rel='self')
+            info_link_tag = OPDSFeed.makeelement("link", **kw)
+            info_links.append(info_link_tag)
+
+        if active_hold and active_hold.integration_client:
```

Comment:
Why check that integration_client is set for holds but not loans?

Reply:
Left over from when there was one annotator that might have a patron or a client, I'll delete it.